### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Wasper**

### CODE REVIEW

File: src/main/java/pl/wasper/bandmanagement/event/dto/EventDto.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import pl.wasper.bandmanagement.serializer.DoubleSerializer;
+import pl.wasper.bandmanagement.serializer.LocalDateTimeDeserializer;
+import pl.wasper.bandmanagement.serializer.LocalDateTimeSerializer;
+
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+public class EventDto {
+    private int id;
```
Comment (gorski) [inline src/main/java/pl/wasper/bandmanagement/event/dto/EventDto.java]: Id is usually type Long

File: src/main/java/pl/wasper/bandmanagement/serializer/DoubleSerializer.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.math.RoundingMode;
+
+public class DoubleSerializer extends JsonSerializer<Double> {
+    @Override
+    public void serialize(Double aDouble, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+            throws IOException, JsonProcessingException {
+        if (aDouble == null) {
+            jsonGenerator.writeNull();
+        } else {
+            BigDecimal bigDecimal = new BigDecimal(aDouble);
```
Comment (gorski) [inline src/main/java/pl/wasper/bandmanagement/serializer/DoubleSerializer.java]: https://stackoverflow.com/questions/12218515/unpredictability-of-the-bigdecimaldouble-constructor

File: src/test/java/pl/wasper/bandmanagement/event/service/EventServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Before
+    public void before() {
+        service = new EventServiceImpl(repository, mapper);
+    }
+
+    @Test
+    public void itShouldReturnEventsList() {
+        List<Event> events = Arrays.asList(
+                FakeFactory.prepareEventWithId(1, LocalDateTime.of(2018, Month.JANUARY, 1, 1, 1), "Gorlice", 1000),
```
Comment (gorski) [inline src/test/java/pl/wasper/bandmanagement/event/service/EventServiceImplTest.java]: if you have more than 2 lines of data, you can use @Parametrized and feed it with array of Objects

File: src/main/java/pl/wasper/bandmanagement/event/dto/EventDto.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @JsonSerialize(using = LocalDateTimeSerializer.class)
+    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
+    private LocalDateTime date;
+    private String place;
+    private String description;
+    @JsonSerialize(using = DoubleSerializer.class)
+    private double price;
+    @JsonSerialize(using = DoubleSerializer.class)
+    private double advance;
+    private boolean isAdvancePaid;
```
[the remark this answers is not in the document]
Comment (Wasper) [reply src/main/java/pl/wasper/bandmanagement/event/dto/EventDto.java]: No, it should be false by default

**Pull request merged**